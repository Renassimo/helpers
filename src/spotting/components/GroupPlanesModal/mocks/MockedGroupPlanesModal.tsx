import { Dispatch, SetStateAction } from 'react';

const MockedGroupPlanesModal = jest.fn(
  ({
    isModalOpen,
    setIsModalOpen,
  }: {
    isModalOpen: boolean;
    setIsModalOpen: Dispatch<SetStateAction<boolean>>;
  }) => (
    <div>
      MockedGroupPlanesModal. isModalOpen: {isModalOpen ? 'open' : 'closed'};
      setIsModalOpen: {!!setIsModalOpen ? 'passed' : 'not passed'}
    </div>
  )
);

export default MockedGroupPlanesModal;
