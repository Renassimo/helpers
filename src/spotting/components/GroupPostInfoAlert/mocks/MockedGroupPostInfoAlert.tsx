const MockedGroupPostInfoAlert = jest.fn(
  ({
    selectedCount,
    onOpenModal,
    clearSelectedIds,
  }: {
    selectedCount: number;
    onOpenModal: () => void;
    clearSelectedIds: () => void;
  }) => (
    <div>
      MockedGroupPostInfoAlert: selectedCount: {selectedCount}; onOpenModal:{' '}
      {!!onOpenModal ? 'passed' : 'not passed'}; clearSelectedIds:{' '}
      {!!clearSelectedIds ? 'passed' : 'not passed'};
    </div>
  )
);

export default MockedGroupPostInfoAlert;
