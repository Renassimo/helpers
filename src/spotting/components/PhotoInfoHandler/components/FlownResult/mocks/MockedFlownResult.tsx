const MockedFlownResult = jest.fn(() => <div>MockedFlownResult</div>);

export default MockedFlownResult;
