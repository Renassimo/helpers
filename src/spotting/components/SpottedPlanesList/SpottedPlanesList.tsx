import React, { useState } from 'react';

import useSpottedPlanes from '@/spotting/hooks/useSpottedPlanes';

import PlanesContainer from '@/spotting/components/PlanesContainer';
import SpottedPlaneCard from '@/spotting/components/SpottedPlaneCard';
import GroupPlanesModal from '@/spotting/components/GroupPlanesModal';

const SpottedPlanesList = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const { spottedPlanes } = useSpottedPlanes();

  return (
    <PlanesContainer>
      {spottedPlanes.map((item) => (
        <SpottedPlaneCard data={item} key={item.id} selectable />
      ))}
      <GroupPlanesModal
        isModalOpen={isModalOpen}
        setIsModalOpen={setIsModalOpen}
      />
    </PlanesContainer>
  );
};

export default SpottedPlanesList;
