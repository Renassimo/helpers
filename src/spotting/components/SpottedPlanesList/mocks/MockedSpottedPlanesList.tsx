const MockedSpottedPlanesList = jest.fn(() => (
  <div>Mocked Spotted Planes List:</div>
));

export default MockedSpottedPlanesList;
