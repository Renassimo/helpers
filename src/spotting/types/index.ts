export * from './spottingTypes';
export * from './spottingDb';
export * from './photoInfoTypes';
