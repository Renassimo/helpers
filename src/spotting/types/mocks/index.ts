export * from './mockedSpottedPlaneApiDataAttributes';
export * from './mockedSpottedPlaneApiData';
export * from './mockedSpottedPlaneCreatedAttributes';
export * from './mockedSpottedPlaneCreatedData';
export * from './spottingDb';
export * from './mockedPhotoInfo';
