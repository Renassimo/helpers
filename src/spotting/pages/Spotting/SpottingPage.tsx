import { useEffect } from 'react';

import { PageInfo, User } from '@/common/types/auth';
import { NotionError } from '@/common/types/notion';
import { SpottedPlaneApiData } from '@/spotting/types';

import SpottingProvider from '@/spotting/providers/SpottingProvider';

import useAlerts from '@/common/hooks/alerts';

import PageTemplate from '@/common/templates/PageTemplate';
import SpottedPlanesList from '@/spotting/components/SpottedPlanesList';

const SpottingPage = ({
  user,
  pages,
  data,
  error,
}: {
  user: User;
  pages: PageInfo[];
  data: SpottedPlaneApiData[] | null;
  error: NotionError | null;
}) => {
  const { createErrorAlert } = useAlerts();

  useEffect(() => {
    if (error) createErrorAlert(error.message || error.code || error.status);
  }, [createErrorAlert, error]);

  return (
    <SpottingProvider data={data}>
      <PageTemplate title="5book" user={user} pages={pages}>
        <SpottedPlanesList />
      </PageTemplate>
    </SpottingProvider>
  );
};

export default SpottingPage;
