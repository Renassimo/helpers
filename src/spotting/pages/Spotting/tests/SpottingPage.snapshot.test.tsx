import renderWithTheme from '@/common/tests/helpers';

import SpottingPage from '@/spotting/pages/Spotting';
import SpottedPlanesList from '@/spotting/components/SpottedPlanesList';
import PageTemplate from '@/common/templates/PageTemplate';

import MockedPageTemplate from '@/common/templates/PageTemplate/mocks';
import MockedSpottedPlanesList from '@/spotting/components/SpottedPlanesList/mocks';
import { mockedSpottedPlaneApiDataTruthy } from '@/spotting/types/mocks';
import { mockedPageInfos, mockedUser } from '@/auth/types/mocks';
import { mockedNotionError418 } from '@/common/types/notion/mocks';

jest.mock('@/spotting/components/SpottedPlanesList');
jest.mock('@/common/templates/PageTemplate');

describe('SpottingPage', () => {
  beforeEach(() => {
    (SpottedPlanesList as unknown as jest.Mock).mockImplementationOnce(
      MockedSpottedPlanesList
    );
    (PageTemplate as unknown as jest.Mock).mockImplementationOnce(
      MockedPageTemplate
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('when got data', () => {
    test('renders successfully', () => {
      // Arrange
      const mockedData = [mockedSpottedPlaneApiDataTruthy];
      // Act
      const { container } = renderWithTheme(
        <SpottingPage
          user={mockedUser}
          pages={mockedPageInfos}
          data={mockedData}
          error={null}
        />
      );
      // Assert
      expect(container).toMatchSnapshot();
      expect(MockedSpottedPlanesList).toHaveBeenCalledWith({}, {});
    });
  });

  describe('when got error', () => {
    test('renders successfully', () => {
      // Arrange
      // Act
      const { container } = renderWithTheme(
        <SpottingPage
          user={mockedUser}
          pages={mockedPageInfos}
          data={null}
          error={mockedNotionError418}
        />
      );
      // Assert
      expect(container).toMatchSnapshot();
      expect(MockedSpottedPlanesList).toHaveBeenCalledWith({}, {});
    });
  });
});
