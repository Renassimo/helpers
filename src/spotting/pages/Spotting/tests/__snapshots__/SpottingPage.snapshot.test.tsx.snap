// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpottingPage when got data renders successfully 1`] = `
<div>
  <div>
    Mocked Page Template (
    Spotting
    ):
    <span>
      Jesse Pinkman
       - 
      5book
    </span>
    <div>
      Mocked Spotted Planes List:
    </div>
  </div>
</div>
`;

exports[`SpottingPage when got error renders successfully 1`] = `
<div>
  <div>
    Mocked Page Template (
    Spotting
    ):
    <span>
      Jesse Pinkman
       - 
      5book
    </span>
    <div>
      Mocked Spotted Planes List:
    </div>
  </div>
</div>
`;
